schema/defs-image.json

```diff
@@ -4,7 +4,7 @@
     "mediaType": {
       "id": "https://opencontainers.org/schema/image/mediaType",
       "type": "string",
-      "pattern": "^[a-z]+/[0-9a-zA-Z.+]+$"
+      "pattern": "^[a-z]+/[0-9a-zA-Z-_+.]+$"
```

Comment:
This is still not quite right.  I've spun off a PR focused on this issue and descriptor tests (pulling in one of your commits from this series) in #448.  Although it would be good if we could get some of these validation PRs landed :p.  Having so many in the air at once is making it hard to keep them orthogonal ;).